torch/csrc/jit/passes/remove_inplace_ops.cpp

```diff
@@ -38,6 +38,22 @@ void ImplicitCastForBinaryInplaceOps(Block* b) {
       if (originalInputs.at(0) == originalInputs.at(1)) {
         continue;
       }
+
+      // Introduce warning for special case where shape values are obtained from
+      // tensor.shape In eager mode these values are python int, so there are
+      // treated as different values. However in tracing both are traced as
+      // tensors, and then they share same memory. This causes a mismatch in
+      // final outputs
+      auto shape_node = originalInputs.at(0)->node();
+      if ((shape_node->kind() == prim::NumToTensor) &&
+          (shape_node->inputs().at(0)->node()->kind() == aten::size)) {
+        std::cerr
```

Comment:
Should we use JIT_LOG rather than write to std::cerr directly?
What are the trade-offs?

Reply:
It seems like  JIT_LOG more for cases such as graph updates/transformations and node deletions. I think it would make more sense to write to std::cerr as we are not necessarily updating any aspect of the graph.